pkg/broker/receiver.go

```diff
@@ -195,10 +195,16 @@ func (r *Receiver) sendEvent(ctx context.Context, tctx cloudevents.HTTPTransport
 	return r.ceClient.Send(sendingCTX, *event)
 }
 
-func (r *Receiver) getTrigger(ctx context.Context, ref types.NamespacedName) (*eventingv1alpha1.Trigger, error) {
+func (r *Receiver) getTrigger(ctx context.Context, ref path.NamespacedNameUID) (*eventingv1alpha1.Trigger, error) {
 	t := &eventingv1alpha1.Trigger{}
-	err := r.client.Get(ctx, ref, t)
-	return t, err
+	err := r.client.Get(ctx, ref.NamespacedName, t)
+	if err != nil {
+		return nil, err
+	}
+	if t.UID != ref.UID {
+		return nil, fmt.Errorf("trigger had a different UID. From ref '%s'. From Kubernetes '%s'", ref.UID, t.UID)
```

Comment:
Optional nitpick: this might be clearer as `Expected ref UID %q and Trigger UID %q to be equal, but were different`